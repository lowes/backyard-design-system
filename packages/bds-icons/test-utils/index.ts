import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
