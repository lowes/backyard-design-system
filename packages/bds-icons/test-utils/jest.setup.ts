import '@testing-library/jest-dom'
import 'jest-styled-components'

jest.setTimeout(1000 * 60 * 20)
