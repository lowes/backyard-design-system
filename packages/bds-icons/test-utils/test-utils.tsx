import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
