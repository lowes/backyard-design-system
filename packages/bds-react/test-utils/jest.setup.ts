import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
