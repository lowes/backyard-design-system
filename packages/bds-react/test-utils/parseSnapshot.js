const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
