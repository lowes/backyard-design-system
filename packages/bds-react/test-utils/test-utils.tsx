import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
