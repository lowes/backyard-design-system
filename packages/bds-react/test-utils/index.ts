// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
