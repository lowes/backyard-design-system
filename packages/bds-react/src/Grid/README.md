# Grid
This component will give you access to the grid container, grid row, and grid columns.

## Installation

```bash
npm install @lowes-tech/bds-react
```

## Usage

```jsx
import { Grid } from '@lowes-tech/bds-react'

<Grid.Container>
    <Grid.Row>
        <Grid.Column></Grid.Column>
        <Grid.Column></Grid.Column>
        <Grid.Column></Grid.Column>
        <Grid.Column></Grid.Column>
    </Grid.Row>
</Grid.Container>
```

## Grid.Container
The container component sets the max width of the container for overall content. This is declared by the container.max token (ie. Lowes.com is set to 1600px) within the Backyard Design System.

## Grid.Row
The row component sets the max width of the row which limits the content width. This is declared by the layout > max_content token (ie. Lowes.com is set to 1356px) within the Backyard Design System. The row component also provides a variety of flex options, passed as classnames, to affect the column layout of the row.

### Flex Classes
Here are a list of the available flex classes that can be passed to the classes prop referenced above

__*Horizontal Alignment*__

| Class | Description |
| :--- | :--- |
| `justify-start` | _items are packed toward the start line_ |
| `justify-center` | _items are packed toward the end line_ |
| `justify-end` | _items are centered along the line_ |
| `justify-around` | _items are evenly distributed in the line with equal space around them_ |
| `justify-between` | _items are evenly distributed in the line; first item is on the start line, last item on the end line_ |
| `justify-evenly` | _items are distributed so that the spacing between any two items (and the space to the edges) is equal_ |

__*Vertical Alignment*__

| Class | Description |
| :--- | :--- |
| `items-stretch` | _stretch to fill the container (still respect min-width/max-width)_ |
| `items-start` | _cross-start margin edge of the items is placed on the cross-start line_ |
| `items-center` | _items are centered in the cross-axis_ |
| `items-end` | _cross-end margin edge of the items is placed on the cross-end line_ |
| `items-baseline` | _items are aligned such as their baselines align_ |

## Grid Column
Grid column is your content container. Each column, unless specifically specified, stretches to fill 100% of the available width divided by the number of columns in the row. (ie. A row with 5 grid columnns would have each grid column take up 20% of the available width.) Each grid column provides you with a 16px padding on the left and right of the column by default. These can be overridden if it does not meet your projects needs.

### Props

| Prop | Type | Description |
| :--- | :--- | :--- |
| `sm` | number | _The number of grid columns the content should span on small (mobile) screens_ |
| `md` | number | _The number of grid columns the content should span on medium (tablets) screens_ |
| `lg` | number | _The number of grid columns the content should span on large (desktop) screens_ |
| `xl` | number | _The number of grid columns the content should span on extra large (large destkop) screens_ |