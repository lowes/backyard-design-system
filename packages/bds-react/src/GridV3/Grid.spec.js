import * as React from 'react'
import { render } from '../../test-utils'
import Grid from './GridV3'

describe('Grid Snapshots', () => {
    test('Grid default snapshot', () => {
        const { asFragment } = render(
            <Grid.Container>
                <Grid.Row>
                    <Grid.Column />
                </Grid.Row>
            </Grid.Container>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
