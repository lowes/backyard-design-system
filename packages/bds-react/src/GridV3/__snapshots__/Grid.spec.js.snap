// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Snapshots Grid default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  box-sizing: border-box;
}

.c0.container--xs {
  margin: 0 auto;
  max-width: 20rem;
}

.c0.container--sm {
  margin: 0 auto;
  max-width: 42rem;
}

.c0.container--md {
  margin: 0 auto;
  max-width: 66rem;
}

.c0.container--lg {
  margin: 0 auto;
  max-width: 82rem;
}

.c0.container--xl {
  margin: 0 auto;
  max-width: 99rem;
}

.c0.container--xxl {
  margin: 0 auto;
  max-width: 117rem;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(var(--cols-default),1fr);
  grid-column-gap: var(--gap-default);
  box-sizing: border-box;
  margin-left: var(--gutter-default);
  margin-right: var(--gutter-default);
}

.c2 {
  box-sizing: border-box;
  width: 100%;
  grid-column-start: var(--col-start);
  grid-column-end: var(--col-end);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: var(--justify-content);
  -webkit-justify-content: var(--justify-content);
  -ms-flex-pack: var(--justify-content);
  justify-content: var(--justify-content);
  justify-items: var(--justify-items);
  -webkit-align-content: var(--align-content);
  -ms-flex-line-pack: var(--align-content);
  align-content: var(--align-content);
  -webkit-align-items: var(--align-items);
  -webkit-box-align: var(--align-items);
  -ms-flex-align: var(--align-items);
  align-items: var(--align-items);
}

@media only screen and (max-width:20rem) {
  .c1 {
    grid-template-columns: repeat(var(--cols-xs),1fr);
    grid-column-gap: var(--gap-xs);
    margin-left: var(--gutter-xs);
    margin-right: var(--gutter-xs);
  }
}

@media only screen  and (min-width:20.0625rem)  and (max-width:42rem) {
  .c1 {
    grid-template-columns: repeat(var(--cols-sm),1fr);
    grid-column-gap: var(--gap-sm);
    margin-left: var(--gutter-sm);
    margin-right: var(--gutter-sm);
  }
}

@media only screen  and (min-width:42.0625rem)  and (max-width:66rem) {
  .c1 {
    grid-template-columns: repeat(var(--cols-md),1fr);
    grid-column-gap: var(--gap-md);
    margin-left: var(--gutter-md);
    margin-right: var(--gutter-md);
  }
}

@media only screen  and (min-width:66.0625rem)  and (max-width:82rem) {
  .c1 {
    grid-template-columns: repeat(var(--cols-lg),1fr);
    grid-column-gap: var(--gap-lg);
    margin-left: var(--gutter-lg);
    margin-right: var(--gutter-lg);
  }
}

@media only screen  and (min-width:82.0625rem)  and (max-width:99rem) {
  .c1 {
    grid-template-columns: repeat(var(--cols-xl),1fr);
    grid-column-gap: var(--gap-xl);
    margin-left: var(--gutter-xl);
    margin-right: var(--gutter-xl);
  }
}

@media only screen  and (min-width:99.0625rem)  and (max-width:117rem) {
  .c1 {
    grid-template-columns: repeat(var(--cols-xxl),1fr);
    grid-column-gap: var(--gap-xxl);
    margin-left: var(--gutter-xxl);
    margin-right: var(--gutter-xxl);
  }
}

<div
    class="c0 container"
  >
    <div
      class="c1 row"
      style="--gutter-default: var(--bds-sizes-size-16); --gutter-xs: var(--bds-sizes-size-16); --gutter-sm: var(--bds-sizes-size-32); --gutter-md: var(--bds-sizes-size-32); --gutter-lg: var(--bds-sizes-size-32); --gutter-xl: var(--bds-sizes-size-40); --gutter-xxl: var(--bds-sizes-size-56); --cols-default: 16; --cols-xs: 4; --cols-sm: 8; --cols-md: 16; --cols-lg: 16; --cols-xl: 16; --cols-xxl: 16; --gap-default: var(--bds-sizes-size-32); --gap-xs: var(--bds-sizes-size-32); --gap-sm: var(--bds-sizes-size-32); --gap-md: var(--bds-sizes-size-32); --gap-lg: var(--bds-sizes-size-32); --gap-xl: var(--bds-sizes-size-32); --gap-xxl: var(--bds-sizes-size-32);"
    >
      <div
        class="c2 col"
      />
    </div>
  </div>
</DocumentFragment>
`;
